### ISSUE TITLE
Create root analogy tester

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#437]
## Goal

Test if root embeddings capture semantic relationships using analogy tasks (e.g., hund:kat :: patr:frat).

## Use Cases

- Validate embedding quality
- Fun demo of semantic structure
- Test specific relationships (family, animals, opposites)
- Documentation/presentation material

## Implementation

Create `scripts/test_root_analogies.py`:

```python
#!/usr/bin/env python3
"""Test semantic analogies with root embeddings."""

import torch
import numpy as np

def solve_analogy(a, b, c, embeddings, root_to_idx, idx_to_root, top_k=5):
    """
    Solve analogy: a:b :: c:?
    
    Returns top-k candidates for d where:
    embedding(d) ≈ embedding(c) + (embedding(b) - embedding(a))
    """
    vec_a = embeddings[root_to_idx[a]]
    vec_b = embeddings[root_to_idx[b]]
    vec_c = embeddings[root_to_idx[c]]
    
    # Vector arithmetic: d ≈ c + (b - a)
    target = vec_c + (vec_b - vec_a)
    
    # Find nearest neighbors to target
    similarities = []
    for root, idx in root_to_idx.items():
        if root in [a, b, c]:  # Exclude input words
            continue
        sim = cosine_similarity(target, embeddings[idx])
        similarities.append((root, sim))
    
    return sorted(similarities, key=lambda x: -x[1])[:top_k]
```

## Test Cases

**Animal Relationships:**
```python
# hund:kat :: hom:? (dog:cat :: human:?)
# Expected: Similar species relationships

# bird:flug :: fiŝ:? (bird:fly :: fish:?)
# Expected: naĝ (swim)
```

**Family Relationships:**
```python
# patr:patrino :: frat:? (father:mother :: brother:?)
# Expected: fratino (sister)

# patr:frat :: patrino:? (father:brother :: mother:?)
# Expected: fratino (sister)
```

**Quality Adjectives:**
```python
# bon:tre :: bel:? (good:very :: beautiful:?)
# Expected: Similar intensification

# grand:mal :: alt:? (large:small :: tall:?)
# Expected: malalt (short) - after affix training
```

**Cognitive Verbs:**
```python
# pens:sci :: leg:? (think:know :: read:?)
# Expected: kompreni (understand)
```

## Features

- [ ] Batch analogy testing from file
- [ ] Accuracy scoring (is correct answer in top-5?)
- [ ] Interactive mode (enter custom analogies)
- [ ] Report generation

## Example Usage

```bash
python scripts/test_root_analogies.py
# Running analogy tests...
# 
# Test: hund:kat :: patr:?
#   1. frat (0.756) ✓
#   2. fil (0.623)
#   3. fratino (0.601)
#
# Test: pens:sci :: leg:?
#   1. kompreni (0.689) ✓
#   2. stud (0.654)
#   3. lern (0.642)
#
# Accuracy: 8/10 (80%)
```

## Deliverables

- [ ] `scripts/test_root_analogies.py`
- [ ] Test suite of 20+ analogies
- [ ] Accuracy report
- [ ] Optional: Add to validation suite

## Success Criteria

- Accuracy > 50% on family relationships
- Accuracy > 40% on semantic fields
- Results make semantic sense (even if not exact)

## Related

- #427 - Root embeddings validation
- #429 - mal- opposition testing (related)
- #430 - Root similarity query tool
